Samples/MedicationsTracker/MedicationsTracker/AddMedicationViewController.swift

```diff
@@ -0,0 +1,156 @@
+    @IBOutlet weak var medicationErrorLabel: UILabel!
+    @IBOutlet weak var doseAmountErrorLabel: UILabel!
+    @IBOutlet weak var nameTableView: UITableView!
+
+    
+    //Mark: Initializers
+    init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?, builder: MedicationBuilder, searcher: MedicationVocabSearcher) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        medicationBuilder = builder
+        self.searcher = searcher
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        doseUnitField.inputView = dosePicker
+        dosePicker.delegate = self
+        nameField.errorLabel = medicationErrorLabel
+        doseAmountField.errorLabel = doseAmountErrorLabel
+        
+        nameTableView.delegate = self
+        nameTableView.dataSource = self
+        nameTableView.isScrollEnabled = true
+        nameTableView.isHidden = true
+    }
+    
+    // MARK: UITextField Delegation
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        // Hide the keyboard.
+        textField.resignFirstResponder()
+        return true
+    }
+    
+    func textFieldShouldEndEditing(_ textField: UITextField) -> Bool {
+        let text = textField as! UIMedicationTextField
+        _ = text.isEmpty()
+        if (textField == doseAmountField){
+            _ = text.isNumeric()
+        }
+        nameTableView.isHidden = true
+
+        return true
+    }
+    
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        let substring = (textField.text! as NSString).replacingCharacters(in: range, with: string)
+        if substring.characters.count >= minSearchSize {
+            nameTableView.isHidden = false
+            searcher?.searchForMeds(searchValue: substring, completion: {autocompleteContents in
+                DispatchQueue.main.async {
+                    self.autoComplete = autocompleteContents
+                }
+            })
+            nameTableView.reloadData()
+        } else{
+            nameTableView.isHidden = true
+        }
+        
+        return true
+    }
+    
+    // MARK: UIPickerView Delegation
+    
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView( _ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return Int(dosePickerData.count)
+    }
+    
+    func pickerView( _ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return dosePickerData[row]
+    }
+    
+    func pickerView( _ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        doseUnitField.text = dosePickerData[row]
+    }
+    
+    // MARK UITableView
```

Comment:
Similarly to the UIPickerTextField, could have a UIAutocompleteTextField that handled the table; then it's in one place too.  There's only one per view now, so wouldn't need to do it yet; but would be helpful if more fields with autocomplete are added.